metagraph/integration_tests/test_assemble.py

```diff
@@ -195,3 +200,103 @@ def test_round_robin_graph_size_via_gfa(self, gfa_test):
         res = subprocess.run(compare_command.split(), stdout=subprocess.PIPE)
         self.assertEqual(res.returncode, 0)
         self.assertEqual("Graphs are identical" in res.stdout.decode().split('\n')[2], True)
+
+@parameterized_class(('graph_repr', 'anno_repr'),
+    input_values=product(
+        [repr for repr in GRAPH_TYPES + ['succinct_bloom', 'succinct_mask'] if not (repr == 'bitmap' and PROTEIN_MODE)],
+        ANNO_TYPES + ['row_diff_brwt_separate',
+                      'row_diff_brwt_no_fork_opt',
+                      'row_diff_brwt_no_anchor_opt']
+    ),
+    class_name_func=get_test_class_name
+)
+class TestDiffAssembly(TestingBase):
+    @classmethod
+    def setUpClass(cls):
+        cls.tempdir = TemporaryDirectory()
+
+        cls.with_bloom = False
+        if cls.graph_repr == 'succinct_bloom':
+            cls.graph_repr = 'succinct'
+            cls.with_bloom = True
+
+        cls.mask_dummy = False
+        if cls.graph_repr == 'succinct_mask':
+            cls.graph_repr = 'succinct'
+            cls.mask_dummy = True
+
+        k = 31
+        construct_command = '{exe} build {mask_dummy} -p {num_threads} \
+                --graph {repr} -k {k} -o {outfile} {input}'.format(
+            exe=METAGRAPH,
+            num_threads=NUM_THREADS,
+            mask_dummy='--mask-dummy' if cls.mask_dummy else '',
+            repr=cls.graph_repr,
+            k=k,
+            outfile=cls.tempdir.name + '/graph',
+            input=TEST_DATA_DIR + '/metasub_fake_data.fa'
+        )
```

Comment:
use f strings

Reply:
There are a lot of places in the integration tests where f strings can be used instead of `format`, so let's save that for a later refactor

Reply:
I looked into this, f-strings only work for variable substitution and don't allow you to compute values within the braces, so I think it's cleaner the way it is now.